src/NuGet.Clients/PackageManagement.VisualStudio/EnvDTEProjectAdapter.cs

```diff
@@ -164,42 +156,21 @@ public bool IsLegacyCSProjPackageReferenceProject
 
                 if (!_isLegacyCSProjPackageReferenceProject.HasValue)
                 {
-                    var restoreProjectStyle = string.Empty;
-
-                    var hasRestoreProjectStyle = _context
-                        .MSBuildProperties
-                        .TryGetValue(ProjectSystemProviderContext.RestoreProjectStyle, out restoreProjectStyle)
-                        && !string.IsNullOrEmpty(restoreProjectStyle);
-
-                    if (!hasRestoreProjectStyle)
+                    // A legacy CSProj can't be CPS, must cast to VSProject4 and *must* have at least one package
+                    // reference already in the CSProj. In the future this logic may change. For now a user must
+                    // hand code their first package reference. Laid out in longhand for readability.
+                    if (AsIVsHierarchy?.IsCapabilityMatch("CPS") ?? true)
                     {
-                        // A legacy CSProj can't be CPS, must cast to VSProject4 and *must* have at least one package
-                        // reference already in the CSProj. In the future this logic may change. For now a user must
-                        // hand code their first package reference. Laid out in longhand for readability.
-                        if (AsIVsHierarchy?.IsCapabilityMatch("CPS") ?? true)
-                        {
-                            _isLegacyCSProjPackageReferenceProject = false;
-                        }
-                        else if (AsVSProject4 == null ||
-                                (AsVSProject4.PackageReferences?.InstalledPackages?.Length ?? 0) == 0)
-                        {
-                            _isLegacyCSProjPackageReferenceProject = false;
-                        }
-                        else
-                        {
-                            _isLegacyCSProjPackageReferenceProject = true;
-                        }
+                        _isLegacyCSProjPackageReferenceProject = false;
                     }
-                    else if (restoreProjectStyle.Equals(ProjectStyle.PackageReference.ToString(), 
-                        StringComparison.OrdinalIgnoreCase))
+                    else if (AsVSProject4 == null ||
+                        (AsVSProject4.PackageReferences?.InstalledPackages?.Length ?? 0) == 0)
```

Comment:
``RestoreProjectStyle`` isn't project system specific, it can be used anywhere. We will need this property for legacy csproj for the same reasons we need it in NETCore. Legacy csproj just doesn't have the same usage yet.